src/search/FindBar.js

```diff
@@ -407,6 +422,41 @@ define(function (require, exports, module) {
     };
 
     /**
+     * @private
+     * Shows the search History in dropdown.
+     */
+    FindBar.prototype.showSearchHints = function () {
+        var self = this;
+        this.$searchField = $("input#find-what");
```

Comment:
This part (L430-L432) with nested `this.$searchField` looks a bit scary, maybe refactor it to use some named variables instead.